Objects/call.c

```diff
@@ -1043,6 +1043,162 @@ _PyStack_UnpackDict_FreeNoDecRef(PyObject *const *stack, PyObject *kwnames)
     Py_DECREF(kwnames);
 }
 
+/* --- Tagged call functions  ------------------------------------------ */
+
+static PyObject *
+PyObject_VectorcallTaggedSlow(PyObject *callable,
+                              const _Py_TaggedObject *tagged, size_t nargsf, PyObject *kwnames)
+{
+    size_t nargs = nargsf & ~PY_VECTORCALL_ARGUMENTS_OFFSET;
+    if (kwnames != NULL && PyTuple_CheckExact(kwnames)) {
+        nargs += PyTuple_GET_SIZE(kwnames);
+    }
+    PyObject **args = PyMem_Malloc((nargs + 1) * sizeof(PyObject *));
+    if (args == NULL) {
+        PyErr_NoMemory();
+        return NULL;
+    }
+    _Py_untag_stack(args + 1, tagged, nargs);
+    PyObject *res = PyObject_Vectorcall(callable, args + 1, nargsf, kwnames);
+    PyMem_Free(args);
+    return res;
+}
+
+PyObject *
+PyObject_Vectorcall_Tagged(PyObject *callable,
+                           const _Py_TaggedObject *tagged, size_t nargsf, PyObject *kwnames)
+{
+#if defined(Py_GIL_DISABLED) || defined(Py_OBJECT_TEST_TAG)
+    size_t nargs = nargsf & ~PY_VECTORCALL_ARGUMENTS_OFFSET;
+    if (kwnames != NULL && PyTuple_CheckExact(kwnames)) {
```

Comment:
Shouldn't `kwnames` always be a `PyTuple_CheckExact()` if it's not `NULL`? In other words, should this be:

```
if (kwnames != NULL) {
    assert(PyTuple_CheckExact(kwnames)));
    ...
}
```